src/features/orders/Timeline.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useCallback, useEffect } from 'react'
+import { Timeline as TimelineComponent, StepProps } from '../../share/timeline'
+import LocalShippingIcon from '@material-ui/icons/LocalShipping'
+import SendIcon from '@material-ui/icons/Send'
+import Icon from '@material-ui/core/Icon'
+import DoneIcon from '@material-ui/icons/Done'
+import { useSelector, useDispatch } from 'react-redux'
+import { selectors, actions } from './reducer'
+import { makeStyles, Theme, useTheme } from '@material-ui/core/styles'
+import DetailLongTextWrapper from './DetailLongTextWrapper'
+import Box from '@material-ui/core/Box'
+import { MAP_STATUS_ID_TO_TEXT } from './constants'
+import RejectedIcon from '@material-ui/icons/Close'
+import { types } from './reducer'
+
+const DISABLE_COLOR = '#9e9e9e';
+
+const useTimelineStyles = makeStyles((theme: Theme) => ({
+    contentWrapper: {
+        width: '35em',
+        [theme.breakpoints.down('xs')]: {
+            width: '15em',
+        },
+    },
+}));
+
+const useDoneStyles = makeStyles({
+    icon: {
+        color: '#0cb500',
+    },
+});
+
+export default function Timeline() {
+    const classes = useTimelineStyles();
+    const dispatch = useDispatch();
+    const theme = useTheme();
+
+    const activeStep = useSelector(selectors.timeline.activeStep);
+    const orderId = useSelector(selectors.detail.id);
+    const order = useSelector(selectors.ordersTable.data)[orderId];
+    const status = order.order_status;
+
+    const handleStepChange = useCallback(
+        (id: number) => {
+            dispatch(actions.timeline.changeActiveStep(id));
+        },
+        [dispatch],
+    );
+
+    useEffect(() => {
+        dispatch(
+            actions.timeline.changeActiveStep(statusToStep(order.order_status)),
+        );
+    }, [order, dispatch]);
+
+    return (
+        <TimelineComponent
+            steps={[
```

Comment:
please extract all logic for steps generation. you already did it partially - you extracted lastStepProps.